spec/lib/ansible/runner/credential/machine_credential_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'ansible/runner'
+require 'ansible/runner/credential'
+
+RSpec.describe Ansible::Runner::MachineCredential do
+  it ".auth_type is the correct Authentication sub-class" do
+    expect(described_class.auth_type).to eq("ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential")
+  end
+
+  context "with a credential object" do
+    around do |example|
+      Dir.mktmpdir("ansible-runner-credential-test") do |dir|
+        @base_dir = dir
```

Comment:
I feel like it was always told to me to never use instance variables with `rspec`, but I don't know that using a `let` with a `nil` default and redefining:

```ruby
let(:base_dir) { nil }

around do |example|
  Dir.mktmpdir("ansible-runner-credential-test") do |dir|
    @base_dir = dir
    example.run
  end
end
```

would make it much cleaner.  ¯\\(°_o)/¯

Reply:
Yeah ... I struggled with this one for a while. I think I prefer it the way it is though.

In fact in your example it looks like you're assuming that `let` utilizes an instance variable of the same name and that doesn't feel like behavior we should rely on.

Reply:
F... I typed the above in a rush... it was supposed to be just `base_dir = dir`.

But yeah, I didn't think it was something you needed to fix.